projects/ng-bulma-lib/src/lib/compenents/carousel/carousel/carousel.component.ts

```diff
@@ -0,0 +1,139 @@
+import { CarouselConfig } from '../carousel-config';
+
+@Component({
+  selector: 'bm-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss']
+})
+
+export class BmCarouselComponent implements OnInit, AfterContentInit, AfterViewInit {
+  @ViewChild('carouselWrapper') carouselWrapper: ElementRef;
+  @ViewChild('carouselContent') carouselContent: ElementRef;
+  @ContentChildren(BmCarouselItemComponent) items: QueryList<BmCarouselItemComponent>;
+  @Input() config: CarouselConfig;
+  carouselContentWidth: number = 0;
+  slidingWidth: number = 0;
+  isNextBtnDisabled: boolean = false;
+  isPrevBtnDisabled: boolean = false;
+  itemContainerWidth: any;
+  autoPlayInterval: any;
+
+  get isLastSlide() {
+    return Math.abs(this.slidingWidth) >= (this.carouselContentWidth - (this.itemContainerWidth * this.config.itemsToShow));
+  }
+
+  constructor() {}
+
+  ngAfterContentInit() {
+    if (this.itemContainerWidth) {
+      this.items.toArray().forEach(item => item.itemContainerWidth = this.itemContainerWidth);
+    }
+  }
+
+  ngAfterViewInit() {
+    this.checkInnerContainerWidth();
+  }
+
+  ngOnInit() {
+    const defaultConfig = this.getDefaultConfigs();
+    this.config = {...defaultConfig, ...this.config};
+    this.itemContainerWidth = Math.ceil(this.carouselWrapper.nativeElement.offsetWidth / this.config.itemsToShow);
+    this.isPrevBtnDisabled = !this.config.infinite;
+
+    if (this.config.itemsToShow < this.config.itemsToScroll) {
+      this.config.itemsToScroll = this.config.itemsToShow;
+    }
+    if (this.config.autoPlay === true) {
+      this.autoPlayFunction();
+    }
+  }
+
+  prevSlide() {
+    if (this.config.infinite === false) {
+      this.isNextBtnDisabled = false;
+    }
+    if (this.config.autoPlay === true) {
+      clearInterval(this.autoPlayInterval);
+      this.autoPlayFunction();
+    }
+
+    if (this.slidingWidth === 0 && this.config.infinite === true) {
+      this.slidingWidth = -(this.carouselContentWidth - (this.itemContainerWidth * this.getItemsToScroll('previous')));
+    } else {
+      this.slidingWidth += (this.itemContainerWidth * this.getItemsToScroll('previous'));
+      this.isPrevBtnDisabled = this.slidingWidth === 0 && this.config.infinite === false;
+    }
+  }
+
+  nextSlide() {
+    if (this.config.infinite === false) {
+      this.isPrevBtnDisabled = false;
+    }
+
+    if (this.config.autoPlay === true) {
+      clearInterval(this.autoPlayInterval);
+      this.autoPlayFunction();
+    }
+
+    if (this.isLastSlide && this.config.infinite === true) {
+      this.slidingWidth = 0;
+    } else {
+      this.slidingWidth -= (this.itemContainerWidth * this.getItemsToScroll('next'));
+      this.isNextBtnDisabled = this.isLastSlide && this.config.infinite === false;
+    }
+  }
+
+  autoPlayFunction() {
+    this.autoPlayInterval = setInterval(() => {
+      this.nextSlide();
+    }, this.config.autoPlaySpeed);
+  }
+
+  checkInnerContainerWidth() {
```

Comment:
Rename: checkInnerContainerWidth => setInnerContainerWidth
And make this `private` function.